Audit logging doc 1205 (#135)
tigergraph/server-docs: modules/troubleshooting/pages/audit-logs.adoc

= Audit Logs

TigerGraph's audit logs track information about user actions such as the action taken, the user permissions, the client used, and so on.

Logging in the same files can also be found for login, and all other requests sent to any of these services.

[NOTE]
ML workbench is an application on top of TigerGraph, so all requests that it sends will be logged in the same manner.

== Query execution via RESTPP

`NGINX#1.out` log contains the endpoint accessed, the IP, and the client.
RESTPP `log.INFO` contains the query run, user who ran it, and the graph it ran against.

> In a cluster, the log for the server that processed the request will contain the details.

Reviewer: ```suggestion
In a cluster, the log for the node that processed the request contains the details.
```

For example, if the request was routed to m3, the log would be in `NGINX#3.out` on m3 or RESTPP `log.INFO` on m3.

Example query:
[source, console]
----
curl -H "Authorization: Bearer 4m6nh1rakn60430rjf5asv" <ip_address>:9000/query/ldbc_snb/example_query
----
Nginx logs:

[source, console]
----
<ip_address> - - [30/Jun/2022:01:37:52 +0000] "GET /query/ldbc_snb/example_query HTTP/1.1" 200 117 "-" "curl/7.29.0"
----

RESTPP logs:

[source, console]
----
I0630 01:41:29.779425 29409 handler.cpp:312] RawRequest|,131076.RESTPP_1_1.1656553289779.N,NNN,0,0,0,S|GET|/query/ldbc_snb/example_query|async = 0|payload_data.size() = 0|api = v2
I0630 01:41:29.779491 29409 handler.cpp:434] RequestInfo|,131076.RESTPP_1_1.1656553289779.N,NNN,0,0,0,S|user:example_user|api:v2|function:queryDispatcher|graph_name:ldbc_snb|libudf:libudf_ldbc_snb
I0630 01:41:29.783354 29410 requestrecord.cpp:349] ReturnResult|0|4ms|GPE_1_1|131076.RESTPP_1_1.1656553289779.N|user:example_user|/query/ldbc_snb/example_query|graph_id=1|117
----

> == Query execution via GraphStudio:

Reviewer: ```suggestion
== Query execution via GraphStudio
```

Nginx `NGINX#1.out` log contains the endpoint accessed, the IP, and the client.
GraphStudio (GUI) `GUI#1.out` contains the query run, user who ran it, and the graph it ran against.

=== Nginx logs
[source, console]
----
136.2.2.2 - - [30/Jun/2022:01:44:58 +0000] "GET /api/restpp/query/ldbc_snb/example_query HTTP/1.1" 202 67 "http://35.2.2.2:14240/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:101.0) Gecko/20100101 Firefox/101.0"
----

=== GraphStudio logs
[source, console]
----
2022-06-30 01:46:28.911 I | middleware/logger.go:63] <-- |     |               | 8d4b9219-0733-4413-8dd2-e4765a7e08da |    example_user |  136.2.2.2 | GET     /api/restpp/query/ldbc_snb/example_query
2022-06-30 01:46:28.922 I | middleware/logger.go:96] --> | 202 |   11.297188ms | 8d4b9219-0733-4413-8dd2-e4765a7e08da |    example_user |  136.2.2.2 | GET     /api/restpp/query/ldbc_snb/example_query
----

> == Query execution via GSQL CLI:

Reviewer: ```suggestion
== Query execution via GSQL CLI
```

> The `NGINX#1.out` log contains the endpoint accessed, the IP, and the client.

Reviewer: Whose IP?
